modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/tpch/TpchTest.java

```diff
@@ -17,22 +17,47 @@
 
 package org.apache.ignite.internal.processors.query.calcite.integration.tpch;
 
+import java.util.Collection;
 import org.apache.ignite.internal.processors.query.calcite.integration.AbstractBasicIntegrationTest;
+import org.apache.ignite.internal.util.typedef.F;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 /** */
-public class TpchQ20Test extends AbstractBasicIntegrationTest {
-    /**
-     * Test the Q20 TPC-H query can be planned and executed.
-     */
-    @Test
-    public void testQ20() throws Exception {
+@RunWith(Parameterized.class)
+public class TpchTest extends AbstractBasicIntegrationTest {
+    /** Query ID. */
+    @Parameterized.Parameter
+    public int qryId;
+
+    /** */
+    @Parameterized.Parameters(name = "queryId={0}")
+    public static Collection<Object> params() {
+        return F.asList(16, 20);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
         TpchHelper.createTables(client);
 
-        TpchHelper.fillTables(client, 0.01);
+        TpchHelper.fillTables(client, 0.1);
 
         TpchHelper.collectSqlStatistics(client);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean destroyCachesAfterTest() {
+        return false;
+    }
 
-        sql(TpchHelper.getQuery(20));
+    /**
+     * Test the TPC-H query can be planned and executed.
+     */
+    @Test
+    public void test() throws Exception {
+        sql(TpchHelper.getQuery(qryId));
```

Comment:
Can also we add something like 
```
        if (log.isInfoEnabled())
            log.info("The test query:\n" + sql);
``` 
in `protected List<List<?>> exec(String sql) `
I think it would be useful for future tpch tickets